src/adapters/streaming-ttl-checker.ts

```diff
@@ -0,0 +1,70 @@
+import { AppComponents } from '../types'
+import { IStreamingChecker } from '../types/checker.type'
+import { CronJob } from 'cron'
+
+export async function createStreamingTTLChecker(
+  components: Pick<AppComponents, 'logs' | 'sceneStreamAccessManager' | 'livekit'>
+): Promise<IStreamingChecker> {
+  const { logs, sceneStreamAccessManager, livekit } = components
+  const logger = logs.getLogger(`streaming-ttl-checker`)
+  let job: CronJob
+  async function start(): Promise<void> {
+    job = new CronJob(
+      '* * * * *',
+      async function () {
+        try {
+          logger.info(`Looking into active streamings.`)
+
+          const activeStreamings = await sceneStreamAccessManager.getActiveStreamings()
+          logger.info(`Found ${activeStreamings.length} active streamings to verify.`)
+
+          const MAX_STREAMING_TIME = 4 * 60 * 60 * 1000 // 4 hours
+          const now = Date.now()
+
+          const expiredStreamings = activeStreamings.filter(
+            (streaming) => now - streaming.created_at > MAX_STREAMING_TIME
+          )
+
+          if (expiredStreamings.length === 0) {
+            return
+          }
+
+          logger.info(`Found ${expiredStreamings.length} streamings that exceed the maximum allowed time.`)
+
+          const ingressIdsToRevoke = expiredStreamings.map((streaming) => streaming.ingress_id)
+
+          await Promise.all(
+            ingressIdsToRevoke.map(async (ingressId) => {
+              try {
+                await livekit.removeIngress(ingressId)
+                await sceneStreamAccessManager.killStreaming(ingressId)
+                logger.info(`Ingress ${ingressId} revoked correctly from LiveKit and streaming killed`)
+              } catch (error) {
+                logger.error(`Error revoking ingress ${ingressId} or killing streaming: ${error}`)
+              }
+            })
+          )
+
+          logger.info(`${expiredStreamings.length} streaming keys revoked for exceeding the maximum allowed time.`)
```

Comment:
If there are 5 ingress to remove and one of them fails, this log will say that 5 streaming keys were revoked but in reality, 4 were removed. We already have the log that occurs when the ingress is revoked correctly, what do you think about removing this one?

Reply:
removed